src/main/java/microsoft/exchange/webservices/data/ExchangeServiceBase.java

```diff
@@ -411,89 +430,29 @@ protected Date convertUniversalDateTimeStringToDate(String dateString) {
     String localPattern1 = "yyyy-MM-dd'Z'";
     String pattern = "yyyy-MM-ddz";
     String localPattern2 = "yyyy-MM-dd'T'HH:mm:ss";
-    DateFormat utcFormatter = null;
     Date dt = null;
-    String errMsg = String.format(
-        "Date String %s not in valid UTC/local format", dateString);
     if (dateString == null || dateString.isEmpty()) {
       return null;
     } else {
       if (dateString.endsWith("Z")) {
-        // String in UTC format yyyy-MM-ddTHH:mm:ssZ
-        utcFormatter = new SimpleDateFormat(utcPattern);
-        utcFormatter.setTimeZone(TimeZone.getTimeZone("UTC"));
-        try {
-          dt = utcFormatter.parse(dateString);
-        } catch (ParseException e) {
-          utcFormatter = new SimpleDateFormat(pattern);
-          utcFormatter.setTimeZone(TimeZone.getTimeZone("UTC"));
-          //	dateString = dateString.substring(0, 10)+"T12:00:00Z";
-          try {
-            dt = utcFormatter.parse(dateString);
-          } catch (ParseException e1) {
-            utcFormatter = new SimpleDateFormat(localPattern1);
-            utcFormatter.setTimeZone(TimeZone.getTimeZone("UTC"));
-
-            try {
-              dt = utcFormatter.parse(dateString);
-            } catch (ParseException ex) {
-
-              utcFormatter = new SimpleDateFormat(utcPattern1);
-              utcFormatter.setTimeZone(TimeZone.getTimeZone("UTC"));
-            }
-            try {
-              dt = utcFormatter.parse(dateString);
-            } catch (ParseException e2) {
-              throw new IllegalArgumentException(errMsg, e);
-            }
-
-          }
-          // throw new IllegalArgumentException(errMsg,e);
-        }
+        dt = parseDateTimeString(dateString, defaultUTC, utcPattern, pattern, localPattern1, utcPattern1);
       } else if (dateString.endsWith("z")) {
-        // String in UTC format yyyy-MM-ddTHH:mm:ssZ
-        utcFormatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'z'");
-        utcFormatter.setTimeZone(TimeZone.getTimeZone("UTC"));
-        try {
-          dt = utcFormatter.parse(dateString);
-        } catch (ParseException e) {
-          throw new IllegalArgumentException(e);
-        }
+        dt = parseDateTimeString(dateString, defaultUTC, "yyyy-MM-dd'T'HH:mm:ss'z'");
       } else {
         // dateString is not ending with Z.
         // Check for yyyy-MM-ddTHH:mm:ss+/-HH:mm pattern
         Matcher localTimeMatcher = localTimePattern.matcher(dateString);
         if (localTimeMatcher.find()) {
-          System.out.println("Pattern matched");
+          log.debug("Pattern matched");
```

Comment:
This logging doesn't really mean anything as it doesn't contain any information about the pattern. It could better be removed, I guess...